WordPress/src/test/java/org/wordpress/android/util/ArrayUtilsTest.kt

```diff
@@ -0,0 +1,96 @@
+package org.wordpress.android.util
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+class ArrayUtilsTest {
+    /* REMOVE */
+
+    @Test
+    fun `given valid index, when removing from start of array, then the result is the expected one`() {
+        val stringArray = arrayOf("one", "two", "three", "four", "five")
+
+        val result = ArrayUtils.remove(stringArray, 0)
+
+        val expected = arrayOf("two", "three", "four", "five")
+        assertThat(result).isEqualTo(expected)
+    }
+
+    @Test
+    fun `given valid index, when removing from middle of array, then the result is the expected one`() {
+        val stringArray = arrayOf("one", "two", "three", "four", "five")
+
+        val result = ArrayUtils.remove(stringArray, 2)
+
+        val expected = arrayOf("one", "two", "four", "five")
+        assertThat(result).isEqualTo(expected)
+    }
+
+    @Test
+    fun `given valid index, when removing from end of array, then the result is the expected one`() {
+        val stringArray = arrayOf("one", "two", "three", "four", "five")
+
+        val result = ArrayUtils.remove(stringArray, 4)
+
+        val expected = arrayOf("one", "two", "three", "four")
+        assertThat(result).isEqualTo(expected)
+    }
+
+    @Test(expected = IndexOutOfBoundsException::class)
+    fun `given invalid negative index, when removing from array, then throw index out of bounds exception`() {
+        val stringArray = arrayOf("one", "two", "three", "four", "five")
+
+        val result = ArrayUtils.remove(stringArray, -1)
+
+        val expected = arrayOf("one", "two", "four", "five")
+        assertThat(result).isEqualTo(expected)
```

Comment:
In my opinion, using [assertFailsWith](https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/assert-fails-with.html) would be more declarative than annotation properties. However, that might be up to personal taste. Either way, may I suggest removing the code that'll not be executed to simplify the test?

```suggestion
        ArrayUtils.remove(stringArray, -1)
```

Reply:
I love this suggestion and I am all in to apply it here @oguzkocer ! 💯 

However, for us to be able to use [assertFailsWith](https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/assert-fails-with.html) in WPAndroid we would first need to add the `org.jetbrains.kotlin:kotlin-test-junit` dependency via `testImplementation`.

Would you like me to proceed with that? 🤔

PS: WCAndroid is already using this dependency, so we could align WPAndroid like that. Yes, WCAndroid is only using that for its `DateUtilsTest` test suite so far, and one could say that it might be better to not introduce an extra such dependencies for that little of a gain, but I don't see any harm with that, at least I can't think of any, especially since this is a test dependency.

Reply:
Partially done: 442e313c668c3dfd0c073e240ec523f4ccc281d9

Reply:
@ParaskP7 I don't think my suggestion here is important enough to add a new dependency. However, I am wondering if other developers stay away from these helpers because they need to add a new dependency. So, maybe it's still worth to add it? 🤔 

I will leave the decision to you as I am good with it either way.

Reply:
Great, I actually agree, thanks for the confirmation on that @oguzkocer !

This is now done: 48002a0f8383b9a4fc586252cc6d895e54938371